func_tests.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+exitcode=0
+
+cd $(dirname "$0")
+for testfile in myapp/tests/test_*.py; do
+	# Running tests separately beacuse web tests can't run together...
+	echo "Running test $testfile"
+	python3 -m unittest $testfile
+	exitcode=$(expr  $? \| "$exitcode")
+done
+
```

Comment:
@murradkh you should not have to write a script like this.  I suggest using a tool like [pytest](https://docs.pytest.org/en/stable/) instead.  it easily lets you run all tests in a directory hierarchy with sophisticated filtering, reporting, code coverage etc.

Reply:
Thank you for your feedback.  We will look into the use of `pytest` instead. I gravitated towards this solution since this is what was provided to us in [the stage one repository ](https://github.com/kamaql97/python-course-kamal). I just attempted to replicate a tried and tested method that integrates well with GitHub (i.e. it displays the test results within the pull request UI).

Reply:
I am now doing it using the GitHub-recommended `.yaml` workflows and I have removed the Linux shell scripts, as per your advice